lib/hsa/mcwamp_hsa.cpp

```diff
@@ -1590,6 +1590,33 @@ class HSAQueue final : public KalmarQueue
     };
 
 
+    void drain() {
+      while (1) {
```

Comment:
I believe comments are needed to clarify the semantic of `drain()`, especially there is a `while(1)` loop here.

Reply:
updated, the exit condition is not controlled by whether a valid is found